tests/test_Register.py

```diff
@@ -0,0 +1,231 @@
+
+            tPoint = afwGeom.Point2D(float(xt), float(yt))
+            iPoint = afwGeom.Point2D(float(xi), float(yi))
+
+            tRecord.set(centroidKey, tPoint)
+            iRecord.set(centroidKey, iPoint)
+            tRecord.set(coordKey, wcs.pixelToSky(tPoint))
+            iRecord.set(coordKey, wcs.pixelToSky(iPoint))
+
+        self.showImage(inputExp, inputSources, "Input", 1)
+        self.showImage(templateExp, templateSources, "Template", 2)
+
+        return Struct(xInput=xInput, yInput=yInput, xTemplate=xTemplate, yTemplate=yTemplate, wcs=wcs,
+                      inputExp=inputExp, inputSources=inputSources,
+                      templateExp=templateExp, templateSources=templateSources)
+
+    def runTask(self, inData, config=RegisterConfig()):
+        """Run the task on the data"""
+        config.sipOrder = 2
+        task = RegisterTask(name="register", config=config)
+        results = task.run(inData.inputSources, inData.inputExp.getWcs(),
+                           inData.inputExp.getBBox(afwImage.LOCAL), inData.templateSources)
+        warpedExp = task.warpExposure(inData.inputExp, results.wcs, inData.templateExp.getWcs(),
+                                      inData.templateExp.getBBox(afwImage.LOCAL))
+        warpedSources = task.warpSources(inData.inputSources, results.wcs, inData.templateExp.getWcs(),
+                                         inData.templateExp.getBBox(afwImage.LOCAL))
+
+        self.showImage(warpedExp, warpedSources, "Aligned", 3)
+        return Struct(warpedExp=warpedExp, warpedSources=warpedSources, matches=results.matches,
+                      wcs=results.wcs, task=task)
+
+    def assertRegistered(self, inData, outData, bad=set()):
+        """Assert that the registration task is registering images"""
+        xTemplate = np.array([x for i, x in enumerate(inData.xTemplate) if i not in bad])
+        yTemplate = np.array([y for i, y in enumerate(inData.yTemplate) if i not in bad])
+        alignedArray = outData.warpedExp.getMaskedImage().getImage().getArray()
+        self.assertTrue((alignedArray[yTemplate, xTemplate] == 1.0).all())
+        for dx in (-1, 0, +1):
+            for dy in range(-1, 0, +1):
+                # The density of points is such that I can assume that no point is next to another.
+                # The values are not quite zero because the "image" is undersampled, so we get ringing.
+                self.assertTrue((alignedArray[yTemplate+dy, xTemplate+dx] < 0.1).all())
+
+        xAligned = np.array([x for i, x in enumerate(outData.warpedSources["center_x"]) if i not in bad])
+        yAligned = np.array([y for i, y in enumerate(outData.warpedSources["center_y"]) if i not in bad])
+        self.assertAlmostEqual((xAligned - xTemplate).mean(), 0, 8)
+        self.assertAlmostEqual((xAligned - xTemplate).std(), 0, 8)
+        self.assertAlmostEqual((yAligned - yTemplate).mean(), 0, 8)
+        self.assertAlmostEqual((yAligned - yTemplate).std(), 0, 8)
```

Comment:
Instead of testing mean and std here, if you derive from `utils.tests.TestCase`, you can use `assertImagesNearlyEqual` to test all the points. And I think you can simplify the `assertTrue(stuff.all())` tests above with `assertFloatsNearlyEqual` too.